ironfish/src/indexer/indexer.ts

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { FileSystem } from '../fileSystems'
+import { IDatabase } from '../storage/database'
+import { createDB } from '../storage/utils'
+
+const DATABASE_VERSION = 1
+
+export abstract class Indexer {
```

Comment:
This is called Indexer, but what logic here is related to indexing? Isn't that all in MinedBlocksIndexer? Most of the abstraction isn't in the indexer.